### PR TITLE
Add methods to start azure node

### PR DESCRIPTION
**Pull request by suchita-g**
Signed-off-by: Suchita Gatfane <sgatfane@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/platform_nodes.py [hunk 1]
```diff
@@ -1674,10 +1674,32 @@ def stop_nodes(self, nodes):
             "Stop nodes functionality is not implemented"
         )
 
-    def start_nodes(self, nodes):
-        raise NotImplementedError(
-            "Start nodes functionality is not implemented"
-        )
+    def start_nodes(self, nodes, timeout=540, wait=True):
+        """Start Azure vm instances
```
Comment (ebenahar) [inline ocs_ci/ocs/platform_nodes.py:1678]: Please move to next line
```python
"""
Start Azure vm instances
```

File: ocs_ci/ocs/platform_nodes.py [hunk 2]
```diff
@@ -1674,10 +1674,32 @@ def stop_nodes(self, nodes):
...
+
+        Args:
+            nodes (list): The OCS objects of the nodes / Azure Vm instance
```
Comment (ebenahar) [inline ocs_ci/ocs/platform_nodes.py:1681]: this arg should be from the same type. 
It should be The OCS objects of the nodes

File: ocs_ci/ocs/platform_nodes.py [hunk 3]
```diff
@@ -1674,10 +1674,32 @@ def stop_nodes(self, nodes):
...
+            logger.error("No nodes found to start")
+            raise ValueError
+        node_names = [n.name for n in nodes]
+        for node_name in node_names:
+            self.azure.start_az_vm_instance(node_name)
+        if wait:
+            """
+            When the node is reachable then 
+            the node reaches status Ready.
+            """
```
Comment (ebenahar) [inline ocs_ci/ocs/platform_nodes.py:1695]: please use `#` here:

```suggestion
            # When the node is reachable then the node reaches status Ready.
```

File: ocs_ci/utility/azure_utils.py [hunk 4]
```diff
@@ -309,3 +309,14 @@ def get_disk_state(self, volume_name):
 
         """
         return self.compute_client.disks.get(self.cluster_resource_group, volume_name).disk_state
+
+    def start_az_vm_instance(self, vm_name):
```
Comment (ebenahar) [inline ocs_ci/utility/azure_utils.py:313]: `az` could be associated with availability zone and not azure. Please change the function name to `start_azure_vm_instance`

File: ocs_ci/utility/azure_utils.py [hunk 5]
```diff
@@ -309,3 +309,14 @@ def get_disk_state(self, volume_name):
...
+        """
+        Start an Azure vm instance
+
+        Args:
+            vm_name: Name of azure vm instance
```
Comment (ebenahar) [inline ocs_ci/utility/azure_utils.py:318]: add an empty line below

File: ocs_ci/utility/azure_utils.py [hunk 6]
```diff
@@ -309,3 +309,14 @@ def get_disk_state(self, volume_name):
...
+        """
+        result = self.compute_client.virtual_machines.start(
+            self.cluster_resource_group, vm_name)
```
Comment (ebenahar) [inline ocs_ci/utility/azure_utils.py:321]:
```suggestion
        result = self.compute_client.virtual_machines.start(
            self.cluster_resource_group, vm_name
        )
```

File: ocs_ci/ocs/platform_nodes.py (hunk 1 above)
Comment (suchita-g) [reply ocs_ci/ocs/platform_nodes.py:1678]: Changed

File: ocs_ci/ocs/platform_nodes.py (hunk 2 above)
Comment (suchita-g) [reply ocs_ci/ocs/platform_nodes.py:1681]: incorporated changes

File: ocs_ci/ocs/platform_nodes.py (hunk 3 above)
Comment (suchita-g) [reply ocs_ci/ocs/platform_nodes.py:1695]: Incorporated changes

File: ocs_ci/utility/azure_utils.py (hunk 4 above)
Comment (suchita-g) [reply ocs_ci/utility/azure_utils.py:313]: Incorporated changes

File: ocs_ci/utility/azure_utils.py (hunk 5 above)
Comment (suchita-g) [reply ocs_ci/utility/azure_utils.py:318]: Incorporated changes

File: ocs_ci/utility/azure_utils.py (hunk 6 above)
Comment (suchita-g) [reply ocs_ci/utility/azure_utils.py:321]: Incorporated changes

**Review by ebenahar — DISMISSED** [review]

**Comment by ebenahar** [comment]
@suchita-g please rebase

File: ocs_ci/ocs/platform_nodes.py [hunk 7]
```diff
@@ -1674,10 +1674,31 @@ def stop_nodes(self, nodes):
...
+
+        Args:
+            nodes (list): The OCS objects of the nodes
+            wait (bool): True for waiting the instances to start, False otherwise
+            timeout (int): time in seconds to wait for node to reach 'ready' state.
+
+        """
+        if not nodes:
+            logger.error("No nodes found to start")
+            raise ValueError
```
Comment (jilju) [inline ocs_ci/ocs/platform_nodes.py:1689]:
```suggestion
            raise ValueError("No nodes found to start")
```

**Review by ebenahar — APPROVED** [review]

**Review by jilju — DISMISSED** [review]

**Review by jilju — APPROVED** [review]

**Review by sidhant-agrawal — APPROVED** [review]

**Review by ebenahar — APPROVED** [review]

**Comment by suchita-g** [comment]
Verified the code in a local branch while running test https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/15779/.
Please add the Verified label.

**Pull request merged**